### ISSUE TITLE
Error: Invalid audience

### ISSUE THREAD

**Issue by Langer75** [robinostlund/volkswagencarnet#131]
Hi

Since two days I got an error while log in and update the vehicle information:

DEBUG:vw_connection:Using service https://msg.volkswagen.de
DEBUG:vw_connection:Initiating new login
DEBUG:vw_connection:Got authorization endpoint
DEBUG:vw_connection:Authenticating with email and password.
DEBUG:vw_connection:Parsing login response.
DEBUG:vw_connection:Login successful, received authorization code.
DEBUG:vw_connection:Trying to fetch user identity tokens.
DEBUG:vw_connection:User identity token verified OK.
INFO:vw_connection:Successfully logged in
DEBUG:vw_connection:Trying to fetch api tokens.
DEBUG:vw_connection:VW-Group API token verified OK.
DEBUG:vw_connection:Fetching vehicles associated with account
DEBUG:vw_connection:HTTP GET "https://msg.volkswagen.de/fs-car/usermanagement/users/v1/VW/DE/vehicles"
DEBUG:vw_connection:Request for "https://msg.volkswagen.de/fs-car/usermanagement/users/v1/VW/DE/vehicles" returned with status code [200]
DEBUG:vw_connection:Found vehicle(s) associated with account.
WARNING:vw_connection:Could not update information: Invalid audience
WARNING:vw_connection:Could not update information: Invalid audience

Does anybody know whats going wrong? It was working for few days. But since two days I got this error.  The App "WeConnect and also myvolkswagen is working as expected.

Thanks in advance

Langer

**Comment by Langer75** [comment]
My python libary installation was somehow corrupt. I have reinstalled the libary and now it works again

**Comment by Langer75** [comment]
as it semms to be a problem for few of us I reopen this issue. As I do not have this problem I can not debug that issue. I can just say where in the code it happens:
in vw_connection.py in dologin in line 127 the update routine (line 523-547) is called. And inside this the except happens. In line 546 the log entry is printed. But I have no idea why. May be somehow is able to debug this line by line

**Comment by birgersp** [comment]
Having the same issue, but only on one of my computers. I have been using my desktop for coding and testing, and then status retrieval has been working fine. Now that I copied the code to my server (a Linux machine) to run it there, I get the same error as discussed here.

I am seeing something that might be relevant. When prompting the server for status, I only get two "instruments" for each of my car (I have two). The instrument is named "request_in_progress". All other "instruments" are marked as "not supported.

Here is a json printout of the data I get during a status update:

```
{
	"state": {},
	"vehicles": [
		{
			"nickname": null,
			"instruments": [
				{
					"state": false,
					"supported": true,
					"attr": "request_in_progress",
					"attributes": {}
				}
			]
		},
		{
			"nickname": null,
			"instruments": [
				{
					"state": false,
					"supported": true,
					"attr": "request_in_progress",
					"attributes": {}
				}
			]
		}
	]
}
```


Really not sure what to make of this. Also the vehicle "nickname" is set to null, for whatever reason.

**Comment by Langer75** [comment]
I am not sure ecause it was some weeks ago. But I think I deleted some files manually. May be that helps?

**Comment by birgersp** [comment]
> I am not sure ecause it was some weeks ago. But I think I deleted some files manually. May be that helps?

No, not really.

**Comment by Langer75** [comment]
uninstall it and look whether all files that belongs to the library were deleted. If not delete it manually

**Comment by birgersp** [comment]
Initially I tested this on Python 3.8. On my Windows PC however I was running Python 3.10. So I installed Python 3.10 on my Linux machine and ran the code in a virtual python env, and then I don't get the Invalid Audience error any more.

**Comment by Madex7** [comment]
After upgrading today, I also got this error. 
Perhaps we can diable the Audience Check? I created a pull request for this
[https://github.com/robinostlund/volkswagencarnet/pull/140](url)

**Comment by milkboy** [comment]
@Madex7 did the update from #140 fix the issue? I had some random jwt issues because of some python-jwt vs pyjwt version changes in my Python 3.9 venv. Ended up recreating, and everything started working.

**Comment by Madex7** [comment]
Yes, the update fixed my issue. I am running the code with python 3.7 on a raspbian OS. An update would probably also fix the issue, because the JWT handling was changed.
But the verification of the signature was already disabled, so I think that disableing the audience check should be no problem and everybody else does not run into this issue anymore.

**Comment by milkboy** [comment]
Fixed by #140
